src/Response/CsvResponse.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-diactoros for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-diactoros/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Diactoros\Response;
+
+use Psr\Http\Message\StreamInterface;
+use Zend\Diactoros\Exception;
+use Zend\Diactoros\Response;
+use Zend\Diactoros\Stream;
+
+use function get_class;
+use function gettype;
+use function is_object;
+use function is_string;
+use function sprintf;
+
+/**
+ * CSV response.
+ *
+ * Allows creating a CSV response by passing a string to the constructor;
+ * by default, sets a status code of 200 and sets the Content-Type header to
+ * text/csv.

```

Comment:
wouldn't be better to accept also/only array with data and process it here?

Reply:
Honestly, hadn't considered that. Thanks.